genesyscloud/task_management_workbin/genesyscloud_task_management_workbin_proxy.go

```diff
@@ -0,0 +1,184 @@
+func (p *taskManagementWorkbinProxy) getTaskManagementWorkbinIdByName(ctx context.Context, name string) (id string, retryable bool, err error) {
+	return p.getTaskManagementWorkbinIdByNameAttr(ctx, p, name)
+}
+
+// getTaskManagementWorkbinById returns a single Genesys Cloud task management workbin by Id
+func (p *taskManagementWorkbinProxy) getTaskManagementWorkbinById(ctx context.Context, id string) (taskManagementWorkbin *platformclientv2.Workbin, statusCode int, err error) {
+	return p.getTaskManagementWorkbinByIdAttr(ctx, p, id)
+}
+
+// updateTaskManagementWorkbin updates a Genesys Cloud task management workbin
+func (p *taskManagementWorkbinProxy) updateTaskManagementWorkbin(ctx context.Context, id string, taskManagementWorkbin *platformclientv2.Workbinupdate) (*platformclientv2.Workbin, error) {
+	return p.updateTaskManagementWorkbinAttr(ctx, p, id, taskManagementWorkbin)
+}
+
+// deleteTaskManagementWorkbin deletes a Genesys Cloud task management workbin by Id
+func (p *taskManagementWorkbinProxy) deleteTaskManagementWorkbin(ctx context.Context, id string) (statusCode int, err error) {
+	return p.deleteTaskManagementWorkbinAttr(ctx, p, id)
+}
+
+// createTaskManagementWorkbinFn is an implementation function for creating a Genesys Cloud task management workbin
+func createTaskManagementWorkbinFn(ctx context.Context, p *taskManagementWorkbinProxy, taskManagementWorkbin *platformclientv2.Workbincreate) (*platformclientv2.Workbin, error) {
+	workbin, _, err := p.taskManagementApi.PostTaskmanagementWorkbins(*taskManagementWorkbin)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create task management workbin: %s", err)
+	}
+
+	return workbin, nil
+}
+
+// getAllTaskManagementWorkbinFn is the implementation for retrieving all task management workbin in Genesys Cloud
+func getAllTaskManagementWorkbinFn(ctx context.Context, p *taskManagementWorkbinProxy) (*[]platformclientv2.Workbin, error) {
+	var allWorkbins []platformclientv2.Workbin
+	pageSize := 200
+
+	for {
+		queryReq := &platformclientv2.Workbinqueryrequest{
+			PageSize: &pageSize,
+		}
+
+		workbins, _, err := p.taskManagementApi.PostTaskmanagementWorkbinsQuery(*queryReq)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get workbin: %v", err)
+		}
+		allWorkbins = append(allWorkbins, *workbins.Entities...)
+
+		// Exit loop if there are no more 'pages'
+		if workbins.After == nil || *workbins.After == "" {
+			break
+		}
+	}
+
+	return &allWorkbins, nil
+}
+
+// getTaskManagementWorkbinIdByNameFn is an implementation of the function to get a Genesys Cloud task management workbin by name
+func getTaskManagementWorkbinIdByNameFn(ctx context.Context, p *taskManagementWorkbinProxy, name string) (id string, retryable bool, err error) {
+	pageSize := 200
+
+	for {
+		queryReq := &platformclientv2.Workbinqueryrequest{
+			PageSize: &pageSize,
+		}
+
+		workbins, _, err := p.taskManagementApi.PostTaskmanagementWorkbinsQuery(*queryReq)
```

Comment:
You can call the function above to get a list of all workbins, then loop through the results to find the name

Reply:
Fixed to use `getAll` method